src/main/java/org/sagebionetworks/web/client/widget/table/modal/fileview/CreateTableViewWizardStep2.java

```diff
@@ -0,0 +1,326 @@
+		keyboardNavigationHandler = ginInjector.createKeyboardNavigationHandler();
+	}
+
+	public void configure(Entity entity, TableType tableType) {
+		configure(entity, tableType, new ArrayList<ColumnModel>());
+	}
+	
+	public void configure(Entity entity, TableType tableType, List<ColumnModel> startingModels) {
+		this.changingSelection = false;
+		this.entity = entity;
+		this.tableType = tableType;
+		this.startingModels = startingModels;
+		resetEditor();
+	}
+
+	@Override
+	public List<ColumnModel> getEditedColumnModels() {
+		return ColumnModelUtils.extractColumnModels(this.editorRows);
+	}
+
+	@Override
+	public boolean validateModel() {
+		return false;
+	}
+
+	@Override
+	public ColumnModelTableRowEditorWidget addNewColumn() {
+		// Create a new column
+		ColumnModel cm = new ColumnModel();
+		cm.setColumnType(ColumnModelsWidget.DEFAULT_NEW_COLUMN_TYPE);
+		cm.setMaximumSize(ColumnModelsWidget.DEFAULT_STRING_MAX_SIZE);
+		// Assign an id to this column
+		ColumnModelTableRowEditorWidget rowEditor = ginInjector.createColumnModelEditorWidget();
+		// bind this row for navigation.
+		if(this.keyboardNavigationHandler != null){
+			this.keyboardNavigationHandler.bindRow(rowEditor);
+		}
+		editor.addColumn(rowEditor);
+		this.editorRows.add(rowEditor);
+		rowEditor.configure(cm, this);
+		checkSelectionState();
+		return rowEditor;
+	}
+	
+
+	@Override
+	public Widget asWidget() {
+		return editor.asWidget();
+	}
+
+	@Override
+	public void onEditColumns() {
+		// reset the editor to the starting state
+		resetEditor();
+	}
+	
+	/**
+	 * Reset the editor.
+	 */
+	public void resetEditor(){
+		// clear the current navigation editor
+		if(this.keyboardNavigationHandler != null){
+			this.keyboardNavigationHandler.removeAllRows();
+		}
+		this.editorRows.clear();
+		editor.configure(ViewType.EDITOR, true);
+		for(ColumnModel cm: this.startingModels){
+			ColumnModelTableRowViewer rowViewer = ginInjector.createNewColumnModelTableRowViewer();
+			ColumnModelUtils.applyColumnModelToRow(cm, rowViewer);
+			rowViewer.setSelectable(true);
+			rowViewer.setSelectionPresenter(this);
+			editor.addColumn(rowViewer);
+			this.editorRows.add(rowViewer);
+		}
+		checkSelectionState();
+	}
+
+	@Override
+	public void setModalPresenter(ModalPresenter presenter) {
+		this.presenter = presenter;
+	}
+	
+	@Override
+	public void onPrimary() {
+		presenter.setLoading(true);
+		// Save it the data is valid
+		if(!validate()){
+			presenter.setErrorMessage(ColumnModelsWidget.SEE_THE_ERROR_S_ABOVE);
+			presenter.setLoading(false);
+			return;
+		}
+		// Get the models from the view and save them
+		List<ColumnModel> newSchema = getEditedColumnModels();
+		synapseClient.setTableSchema(entity, newSchema, new AsyncCallback<Void>(){
+			@Override
+			public void onFailure(Throwable caught) {
+				presenter.setErrorMessage(caught.getMessage());
+			}
+			
+			@Override
+			public void onSuccess(Void result) {
+				presenter.onFinished();
```

Comment:
Column model editor should show Facet column for "FileView" type (services will be changed)